lib/Runtime/Library/DateImplementation.cpp

```diff
@@ -1484,7 +1484,9 @@ namespace Js {
             continue;
         }
 
-        if (lwNil == lwYear || lwNil == lwMonth || lwNil == lwDate)
+        if (lwNil == lwYear ||
+            lwNil == lwMonth || lwMonth > 11 ||
+            lwNil == lwDate || lwDate > 31)
```

Comment:
How does this handle months with less than 31 days? In a quick test, Firefox does not accept "2017-2-29" (while it _does_ accept "2016-2-29", since 2016 was a leap year) while I believe this would accept both and overflow the first version. I am not sure if that's a dealbreaker or if it is even feasible with all of the custom logic we have here, just pointing it out.

Reply:
@jackhorton see discussion #4329 -- basically it's a lot of extra cost to figure out exactly how many days are in a given `mm/yy` because of leap year rules. d8 fails only for months >12 and days >31, and this PR will align with their behavior, bringing the web reality into 50/50 on this point, and the behavior is not so bizarre as to completely defy user expectations.